lib/miq_automation_engine/engine/miq_ae_workspace.rb

```diff
@@ -54,14 +54,24 @@ def initialize(options = {})
       @current_state_info = {}
       @state_machine_objects = []
       @ae_user = nil
+      @rbac    = false
     end
 
     def readonly?
       @readonly
     end
 
+    def self.current=(ws)
+      Thread.current.thread_variable_set(:current_workspace, ws)
+    end
+
+    def self.current
+      Thread.current.thread_variable_get(:current_workspace)
+    end
+
     def self.instantiate(uri, user, attrs = {})
       workspace = MiqAeWorkspaceRuntime.new(attrs)
+      self.current = workspace
```

Comment:
not sure what I think about this being in here.
You are setting the `current_user` outside of here for some reason.
So I'd imagine the same logic would apply for you calling this externally.
(I can't remember the reasoning)